tests/Graph/AllShortestPathsFinderTest.cpp

```diff
@@ -0,0 +1,644 @@
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Anthony Mahanna
+////////////////////////////////////////////////////////////////////////////////
+
+#include "gtest/gtest.h"
+
+#include "./MockGraph.h"
+#include "./MockGraphProvider.h"
+
+// Used for StringUtils size_t variant
+#include "Basics/operating-system.h"
+
+#include "../Mocks/Servers.h"
+
+#include "Aql/Query.h"
+#include "Basics/GlobalResourceMonitor.h"
+#include "Basics/ResourceUsage.h"
+#include "Basics/StaticStrings.h"
+#include "Basics/StringUtils.h"
+#include "Graph/Enumerators/TwoSidedEnumerator.h"
+#include "Graph/Options/TwoSidedEnumeratorOptions.h"
+#include "Graph/PathManagement/PathStore.h"
+#include "Graph/Queues/FifoQueue.h"
+
+// Needed in case of enabled tracing
+#include "Graph/PathManagement/PathStoreTracer.h"
+#include "Graph/Queues/QueueTracer.h"
+#include "Graph/algorithm-aliases.h"
+
+#include <velocypack/HashedStringRef.h>
+
+using namespace arangodb;
+using namespace arangodb::graph;
+using namespace arangodb::velocypack;
+
+namespace arangodb {
+namespace tests {
+namespace graph {
+
+class AllShortestPathsFinderTest
+    : public ::testing::TestWithParam<MockGraphProvider::LooseEndBehaviour> {
+  using AllShortestPathsFinder = AllShortestPathsEnumerator<MockGraphProvider>;
+  // using AllShortestPathsFinder = TracedKPathEnumerator<MockGraphProvider>;
+
+ protected:
+  bool activateLogging{false};
+  MockGraph mockGraph;
+  mocks::MockAqlServer _server{true};
+  std::shared_ptr<arangodb::aql::Query> _query{_server.createFakeQuery()};
+  arangodb::GlobalResourceMonitor global{};
+  arangodb::ResourceMonitor resourceMonitor{global};
+
+  // PathValidatorOptions parts (used for API not under test here)
+  aql::Variable _tmpVar{"tmp", 0, false};
+  arangodb::aql::AqlFunctionsInternalCache _functionsCache{};
+
+  arangodb::transaction::Methods _trx{_query->newTrxContext()};
+  arangodb::aql::FixedVarExpressionContext _expressionContext{
+      _trx, *_query.get(), _functionsCache};
+  AllShortestPathsFinderTest() {
+    if (activateLogging) {
+      Logger::GRAPHS.setLogLevel(LogLevel::TRACE);
+    }
+
+    /* a chain 1->2->3->4->5 with shortcuts */
+    mockGraph.addEdge(1, 2);
+    mockGraph.addEdge(2, 3);
+    mockGraph.addEdge(3, 4);
+    mockGraph.addEdge(4, 5);
+    mockGraph.addEdge(1, 3);
+    mockGraph.addEdge(1, 4);
+    mockGraph.addEdge(3, 5);
+
+    /* a hexagon 6->7->8->9->10->11->6 */
+    mockGraph.addEdge(6, 7);
+    mockGraph.addEdge(7, 8);
+    mockGraph.addEdge(8, 9);
+    mockGraph.addEdge(9, 10);
+    mockGraph.addEdge(10, 11);
+    mockGraph.addEdge(11, 6);
+    // TODO Anthony: Figure out how to set direction to ANY (looks like default
```

Comment:
Are those TODOs still open?

The edge on the graph is always outbound, I think we need to look into the Index calculation.
I think for the mock graph that is indeed only outbound for now.

Reply:
I do not request to change anything here.

Reply:
these were also out of date, addressed in 7dd39c35d94